config/config.go

```diff
@@ -1264,21 +1264,18 @@ func (cfg *InstrumentationConfig) ValidateBasic() error {
 	if cfg.PyroscopeTrace && cfg.PyroscopeURL == "" {
 		return errors.New("pyroscope_trace can't be enabled if profiling is disabled")
 	}
-	// if there is not InfluxURL configured, then we do not need to validate the rest
+	// if there is not TracePushConfig configured, then we do not need to validate the rest
 	// of the config because we are not connecting.
-	if cfg.InfluxURL == "" {
+	if cfg.TracePushConfig == "" {
 		return nil
 	}
-	if cfg.InfluxToken == "" {
+	if cfg.TracePullAddress == "" {
 		return fmt.Errorf("token is required")
```

Comment:
[blocking] this error looks incredibly misleading given the field rename

```suggestion
		return fmt.Errorf("trace pull address is required")
```

On second thought, is this field optional or required? I suspect it's optional in which case the whole conditional can be deleted.

Reply:
Copying code from elsewhere in this PR:

```
# The tracer pull address specifies which address will be used for pull based
# event collection. If empty, the pull based server will not be started.
```

seems optional so we shouldn't error here